### PR TITLE
feat(tree-view): remove shim and fallbacks

### PR DESCRIPTION
**Pull request by valentin-mladenov**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [x] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: CDE-1226

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: projects/angular/src/utils/_mixins.scss [hunk 1]
```diff
@@ -36,9 +36,9 @@ $clr-outline-spread: variables.$clr_baselineRem_2px;
   outline-offset: variables.$clr_baselineRem_1px;
 }
 
-@mixin include-slim-outline-style-form-fields() {
-  outline: Highlight solid variables.$clr_baselineRem_2px;
-  outline: -webkit-focus-ring-color solid variables.$clr_baselineRem_2px;
+@mixin include-slim-outline-style-form-fields($width: variables.$clr_baselineRem_2px) {
+  outline: Highlight solid $width;
+  outline-color: -webkit-focus-ring-color;
 }
```
Comment (dtsanevmw) [inline projects/angular/src/utils/_mixins.scss:42]: We should use cds token and separate the properties the way we wanted due to the issue found with custom properties.
```suggestion
@mixin include-slim-outline-style-form-fields($width: tokens.$cds-global-space-2) {
  outline-color: Highlight;
  outline-color: -webkit-focus-ring-color;
  outline-style: solid;
  outline-width: $width;
}
```

File: projects/angular/src/data/tree-view/_properties.tree-view.scss [hunk 2]
```diff
@@ -4,38 +4,43 @@
...
+    --clr-tree-link-hover-color: #{tokens.$cds-alias-object-interaction-background-hover};
+    // @deprecated in v17 in favor of clr-tree-node-content-bg-selected-color
+    --clr-tree-link-selection-color: #{tokens.$cds-alias-object-interaction-background-selected};
+    // @deprecated in v17 in favor of clr-tree-node-content-color
+    --clr-tree-link-text-color: #{tokens.$cds-alias-object-interaction-color};
+
+    --clr-tree-node-caret-link-color: var(--clr-tree-node-caret-color);
+    --clr-tree-node-caret-link-hover-color: #{tokens.$cds-alias-object-interaction-color-hover};
+    --clr-tree-node-caret-link-active-color: #{tokens.$cds-alias-object-interaction-color-active};
+    --clr-tree-node-caret-link-disabled-color: #{tokens.$cds-alias-object-interaction-color-disabled};
```
Comment (dtsanevmw) [inline projects/angular/src/data/tree-view/_properties.tree-view.scss:27]: Do we really need these to be named like that? The class which is using these is `clr-treenode-link` so the name `caret-link-color` sound like setting the focus on the caret while it should be the link

**Review by dtsanevmw — COMMENTED** [review]
Two suggestions. We can sync on them

File: projects/angular/src/utils/_mixins.scss (hunk 1 above)
Comment (valentin-mladenov) [reply projects/angular/src/utils/_mixins.scss]: Corrected in commit 59b8c82

**Review by dtsanevmw — APPROVED** [review]

**Pull request merged**